spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/EurekaDiscoveryClientConfigServiceBootstrapConfiguration.java

```diff
@@ -70,6 +71,13 @@ public final void onApplicationEvent(final ApplicationReadyEvent event) {
 			for (String name : beanNamesForType) {
 				registry.removeBeanDefinition(name);
 			}
+
+
+			beanNamesForType = context
+					.getBeanNamesForType(ReactiveDiscoveryClient.class);
+			for (String name : beanNamesForType) {
+				registry.removeBeanDefinition(name);
```

Comment:
Why are you doing this?

Reply:
Remove a and B to recreate the beans and use the new EurekaClient

Reply:
Reference resources https://github.com/spring-cloud/spring-cloud-netflix/issues/3675#issuecomment-579771738

Reply:
When spring.cloud.config.discovery.enabled=true two EurekaClient will be started.
The first EurekaClient is used to get the URL of the configuration service during the bootstrap phase and then close this EurekaClient. (This object is abbreviated to bootstrapEurekaClient) This phase has generated the beans for DiscoveryClient (this object is abbreviated to bootstrapDiscoveryClient) and ReactiveDiscoveryClient (this object is abbreviated to bootstrapReactiveDiscoveryClient).
The second EurekaClient is used to register and access services normally with the registry. (hereinafter referred to as eurekaClient1), and also generates beans corresponding to DiscoveryClient (hereinafter referred to as discoveryClient1) and ReactiveDiscoveryClient (hereinafter referred to as reactiveDiscoveryClient1).

Due to the late closure of bootstrapEurekaClient in the original code, the beans of bootstrapDiscoveryClient and ReactiveDiscoveryClient were not removed in the context.
Cause bootstrapDiscoveryClient and ReactiveDiscoveryClient to be injected at the time of application startup. Since bootstrapEurekaClient is closed, a new list of services is not available.

My added code removes the bootstrapDiscoveryClient and bootstrapReactiveDiscoveryClient beans at the end of the bootstrap phase, causing the program to inject discoveryClient1 and activeDiscoveryClient1 at boot time.

Reply:
@k631583871 We are going to have a discussion with the Netflix Eureka team and see if they could support a bootstrap registration mode so that we don't need to remove beans in this way.